### DISCUSSION TITLE
Adding private instance to Firefox (and FF derivatives)

### DISCUSSION THREAD

**Discussion by cameronj86** [searxng/searxng#3145] [Show and tell]
Scoured the internet and found 1-2 drips of info but it was poorly documented. IME, this method is ideal if:
- **You are using a Firefox derivative that doesn't have an 'add search engine to Firefox' prompt that shows up in the address bar**
- **You connect to an unmasked IP address and your port # isn't being included when you select 'add search engine to Firefox'** 

1. Do a search in your self-hosted instance for the word 'test'. To the far right of the results, click on 'search url' and copy the url. It'll look something like this: [https://192.168.12.001/search?q=test&language=auto&time_range=&safesearch=0&categories=general](https://192.168.12.1/search?q=test&language=auto&time_range=&safesearch=0&categories=general) For brevity's sake, I'll truncate everything after 'test' in the examples @ step 10
2. Download [this](https://addons.mozilla.org/en-US/firefox/addon/mozlz4-edit/) extension to make the mozlz4 file readable.
3. In FF, go to **about:support** to get your profile directory (if using flatpak, it's located in ~/home/.var/app/ )
4. Find the **search.json.mozlz**4 file w/i your directory. Write down the location or take a screenshot of it
5. Launch the downloaded extension. (Should open in a new tab)
6. Click 'open file' and open the **search.json.mozlz4** file.
7. Select **Add Engines**
8. For the next few steps I deleted the info for Yandex and replaced it w/ my searxng instance in steps 9 - 12.
9. Replace the first line w/ a name for your private searxng instance
10. For the second line, paste the url string from step 0 and make the following edits:
11. **Big step** Manually add your port. For example if you're using port 1234 [https://192.168.12.001/search?q=test](https://192.168.12.1/search?q=test) becomes [https://192.168.12.001:1234/search?q=test](https://192.168.12.1:1234/search?q=test)
12. **Big step** If you aren't using SSL, then change https to http. So [https://192.168.12.001:1234/search?q=test](https://192.168.12.1:1234/search?q=test) becomes [http://192.168.12.001:1234/search?q=test](http://192.168.12.1:1234/search?q=test)
13. **Big step** Replace 'test' w/ {searchTerms}. So, w/ all those edits together, it'll look like this: [http://192.168.12.001:1234/search?q={searchTerms}](http://192.168.12.1:1234/search?q={searchTerms})
14. For the last line, you can download the searxng icon and point to it
15. Click the add button
16. Restart Firefox and the custom search engine should show up as an option in the 'search' section of settings.

Thumbs up if it works, comment if you are having trouble or think a step can be better worded

**Comment by unixfox** [comment]
Big warning: this is unsupported officially and the official way is to simply add the search engine like normally.

This will use opensearch if the instance is correctly configured.

**Comment by cameronj86** [comment]
Woof. Today I realized you can use the libredirect extension and just paste your private instance's IP address & port.
